scripts/target_config_lib.lua

```diff
@@ -102,20 +86,25 @@ local function feature_packages(features)
 	return feature_lib.get_packages(files, features)
 end
 
--- This involves running a few processes to evaluate site.mk, so we add a simple cache
-local class_cache = {}
-local function class_packages(class)
-	if class_cache[class] then
-		return class_cache[class]
-	end
+local function site_specific_packages(dev_info)
+	local site_packages
+	local feature_inherited_pkgs
+	local site_features
+
+	-- First read enabled features from site
+	site_features = image_customization_lib.get_selection('feature', env, dev_info)
+	site_features = compact_list(site_features, false)
+
+	-- Create List from packages inherited from features
+	feature_inherited_pkgs = feature_packages(site_features)
 
-	local features = site_vars(string.format('$(GLUON_FEATURES) $(GLUON_FEATURES_%s)', class))
-	features = compact_list(split(features), false)
+	-- Read list of packages from site
+	site_packages = image_customization_lib.get_selection('package',  env, dev_info)
```

Comment:
We are getting the selections of `feature` and `package` right next to each other. It would be more efficient to have a `get_selections`, so `evaluate_device` needs to run only once here.